web_monitoring/html_diff_render.py

```diff
@@ -187,8 +187,16 @@
 MAX_SPACERS = 2500
 
 
+class Strict_url_rule:
+    mode = False
+    REGEX_RULES = {'WBM': r"/web/\d{14}",
+                   'jsessionid': r";jsessionid[^;=]*=([^;]+)",
+                   'UKWA': r"https://www\.webarchive\.org\.uk/wayback/en/archive/\d{14}mp_/"}
```

Comment:
Performance improvement idea:

Use `re.compile(r"/web/\d{14}")` here to compile regex initially and store them in the `REGEX_RULES` dict.

Then, later do:
```
        match_element = Strict_url_rule.REGEX_RULES[Strict_url_rule.mode].search(element)
```
Instead of what you currently do:
```
        match_element = re.search(Strict_url_rule.REGEX_RULES[Strict_url_rule.mode], element)
```
It will be MUCH faster.